src/app/wallet/wallet.component.html

```diff
@@ -43,94 +43,160 @@ <h4 class="mb-3 text-uppercase">
-                            </table>
+                    <div *ngIf="transactionHistory.length === 0">
+                        <span class="transactions__empty"></span>Your transaction history is empty.
+                    </div>
+                    <div *ngIf="transactionHistory !== undefined && transactionHistory.length > 0" 
+                         class="wow fadeIn">
+                        <div class="transactions__history">
+                            <ng-container *ngFor="let transaction of transactionHistoryPage">
+                                <ng-container *ngIf="shouldShowTransaction(transaction)">
+                                    <div class="transactions__single-element" 
+                                         [class.transactions__error] = "transaction.state !== 'MINED' || transaction.type === 'APPROVE_INVESTMENT'">
+                                        <div class="transactions__left">
+                                            <div class="left__desc">
+                                                <div class="left__transaction-type" 
+                                                    *ngIf="transaction.type === 'INVEST'">
+                                                    Investment <span class="left__conjuctor">to</span>
+                                                </div>
+                                                <div class="left__transaction-type" 
+                                                    *ngIf="transaction.type === 'DEPOSIT'">
+                                                    Deposit
+                                                </div>
+                                                <div class="left__transaction-type" 
+                                                    *ngIf="transaction.type === 'WITHDRAW'">
+                                                    Withdraw
+                                                </div>
+                                                <div class="left__transaction-type" 
+                                                    *ngIf="transaction.type === 'SHARE_PAYOUT'">
+                                                    Dividend <span class="left__conjuctor">from</span>
+                                                </div>
+                                                <div class="left__transaction-type" 
+                                                    *ngIf="transaction.type === 'CANCEL_INVESTMENT'">
+                                                    Cancel investment <span class="left__conjuctor">to</span>
+                                                </div>
+                                                <div class="left__transaction-type" 
+                                                    *ngIf="transaction.type === 'APPROVE_INVESTMENT'">
+                                                    Approve Investment <span class="left__conjuctor">to</span>
+                                                </div>
```

Comment:
too much code for a simple mapping that can be done in a pipe.

Reply:
I agree, wanted to ask you if there was any other way to write this logic without all of the unnecessary conditions but I forgot 😄